build_exercise_proj/CMakeLists.txt

```
@@ -1,24 +1,25 @@
-cmake_minimum_required(VERSION 3.1)
-project(build_exercise)
+cmake_minimum_required(VERSION 3.1) 
+project(build_exercise) 
 
-set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON) 
 
-set(_INC_FOLDER_NAME include)
-set(_INC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/${_INC_FOLDER_NAME})
-set(_SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src)
+set(_INC_FOLDER_NAME include) 
+set(_INC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/${_INC_FOLDER_NAME}) 
+set(_SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src) 
 
 
-add_executable(
-    ${PROJECT_NAME}     
-        ${_SRC_DIR}/main.cpp
+add_executable( 
+    ${PROJECT_NAME}      
+        ${_SRC_DIR}/main.cpp 
 )
 
-add_subdirectory(tools_lib_proj)
+add_subdirectory(tools_lib_proj) 
 
-target_include_directories(${PROJECT_NAME} PUBLIC tools_lib_proj)
+target_include_directories(${PROJECT_NAME} PUBLIC tools_lib_proj) 
```

It is a good idea for every project to have a dedicated "include" and "src" section.
You've icluded the "src" section, but not the "include" one.
When you introduce an "include" section, then the call to target_include_directories will include precisely that directory.

```suggestion
target_include_directories(${PROJECT_NAME} PUBLIC ${_INC_DIR}) 
```